node/src/BaseClient.ts

```diff
@@ -348,20 +348,63 @@ function convertGlideRecordForZSet(
 
 /**
  * @internal
- * Downcast `GlideRecord` to `Record`. Use if you are 146% aware that `data` keys are always strings.
+ * Recursevly downcast `GlideRecord` to `Record`. Use if `data` keys are always strings.
  */
 export function glideRecordToRecord<T>(
     data: GlideRecord<T>,
 ): Record<string, T> {
     const res: Record<string, T> = {};
 
     for (const pair of data) {
-        res[pair.key as string] = pair.value;
+        let newVal = pair.value;
+
+        if (isGlideRecord(pair.value)) {
+            newVal = glideRecordToRecord(
+                pair.value as GlideRecord<unknown>,
+            ) as T;
+        } else if (isGlideRecordArray(pair.value)) {
+            newVal = (pair.value as GlideRecord<unknown>[]).map(
+                glideRecordToRecord,
+            ) as T;
+        }
+
+        res[pair.key as string] = newVal;
     }
 
     return res;
 }
 
+/**
+ * @internal
+ * Check whether an object is of type `GlideRecord`.
+ */
+function isGlideRecord(obj?: unknown): boolean {
+    return (
+        obj !== undefined &&
+        obj !== null &&
+        Array.isArray(obj) &&
+        obj.length > 0 &&
+        typeof obj[0] === "object" &&
+        !Array.isArray(obj[0]) &&
+        "key" in obj[0] &&
+        "value" in obj[0]
+    );
+}
+
+/**
+ * @internal
+ * Check whether an object is of type `GlideRecord[]`.
+ */
+function isGlideRecordArray(obj?: unknown): boolean {
+    return (
```

Comment:
does this work instead? You should be able to check undefined, nulls, length via the truth check: 
```
return obj && Array.isArray(obj) && isGlideRecord(obj[0]);
```